packages/web-client/src/webdav/listFiles.ts

```diff
@@ -1,20 +1,26 @@
 import { buildResource } from 'files/src/helpers/resources'
-import { DavProperties } from 'web-pkg/src/constants'
+import { DavProperties, DavProperty } from 'web-pkg/src/constants'
 import { isPublicSpaceResource, Resource, SpaceResource } from '../helpers'
 import { WebDavOptions } from './types'
 
+export type ListFilesOptions = {
+  depth?: number
+  davProperties?: DavProperty[]
+}
+
 export const ListFilesFactory = ({ sdk }: WebDavOptions) => {
   return {
     async listFiles(
       space: SpaceResource,
-      { path, depth = 1 }: { path?: string; depth?: number }
+      { path }: { path?: string },
+      { depth = 1, davProperties }: ListFilesOptions = {}
     ): Promise<Resource[]> {
       let resources: Resource[]
       if (isPublicSpaceResource(space)) {
         resources = await sdk.publicFiles.list(
           `${space.webDavPath.replace(/^\/public-files/, '')}/${path || ''}`,
           space.publicLinkPassword,
-          DavProperties.PublicLink,
+          [DavProperties.PublicLink].concat(davProperties),
```

Comment:
Does this make sense? Do we always want to add this? Is it harmful to pass it to private files? 
If we don't add it by default, I'm wondering how API consumers are supposed to add it if they need it?

Reply:
no, if you provide dav properties they should completely replace the default dav properties. Otherwise it would not be possible to do a propfind with just the id prop for example. All other requested data would be useless and bloat the response payload.

Reply:
Should we remove the PublicLink property from davProperties for non PublicLink spaces?
Or doesn't it harm to request it even if it does not make sense?

I'm think of requesting additional properties for a resource and not having to think about whether we're dealing with a public link... does that make sense at all? I haven't worked with the webdav api really, so it's hard for me to judge...

Reply:
I've replaced the code with `davProperties || [DavProperties.PublicLink]`. Still wondering, should we filter the PublicLink property out for the private files?